File: src/Classes/Campaign/Campaign.ts
import { CampaignNote } from "../../../global";
import Character from "../Character/Character";
import Encounter from "../Encounter/Encounter";

export default class Campaign {
    id: number;
    name: string;
    description: string;
    notes: Array<CampaignNote>;
    encounters: Array<Encounter>;
    characters: Array<Character>;
    [key: string]: any;

    constructor(id: number, name: string, description: string, notes: Array<CampaignNote>,
        encounters: Array<Encounter>, characters: Array<Character>) {
            this.id = id;
            this.name = name;
            this.description = description;
            this.notes = notes;
            this.encounters = encounters;
            this.characters = characters;
    }
}

File: Models/Campaign.cs

using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace dnd_buddy.Models
{
    public class Campaign {
        [Key]
        public int Id {get; set;}
        public string Name {get; set;}
        public string Description {get; set;}
        public List<CampaignNote>? Notes {get; set;}
        [ForeignKey("CampaignId")]
        public virtual List<Encounter>? Encounters {get; set;}
        [ForeignKey("CampaignId")]
        public List<Character>? Characters {get; set;}
        public Campaign() {
            this.Encounters = new List<Encounter>();
            this.Name = "New Campaign";
            this.Description = "";
            this.Characters = new List<Character>();
            this.Notes = new List<CampaignNote>();
        }
    }
}